### PR TITLE
Expose UI support for custom fields on financial types

### PR DESCRIPTION
**Pull request by pradpnayak**
Overview
----------------------------------------
https://lab.civicrm.org/dev/financial/issues/29

Before
----------------------------------------
Custom fields were not supported for Financial Type

After
----------------------------------------
after

### CODE REVIEW

**Comment by pradpnayak** [comment]
Can anyone review this for me?

**Comment by eileenmcnaughton** [comment]
I haven't reviewed but I peeked at the code & the coding approach taken here is consistent with the work we have been doing in this area.

**Comment by eileenmcnaughton** [comment]
@pradpnayak the one issue I have with this is that you are enabling custom fields out of the box. My position so far has been that we should make it possible for extensions to enable these but not do it in core.

My main concern with enabling in core is it sets expectations that we then accept things like making the data available in searches / reports etc & that is a whole extra conversation. I think we should remove that part before we merge it & discuss separately (ok to leave in for now as it might make it easier for a reviewer)

File: templates/CRM/Financial/Form/FinancialType.tpl [hunk 1]
```diff
@@ -24,46 +24,9 @@
...
-    </div>
-   {/if}
-</div>
-
+{include file="CRM/Core/Form/EntityForm.tpl"}
+{if $action eq 2 or $action eq 4 } {* Update or View*}
+  <div class="crm-submit-buttons">
+    <a href="{crmURL p='civicrm/admin/financial/financialType/accounts' q="action=browse&reset=1&aid=$aid"}" class="button"><span>{ts}View or Edit Financial Accounts{/ts}</a></span>
+  </div>
+{/if}
```
Comment (pradpnayak) [inline templates/CRM/Financial/Form/FinancialType.tpl]: @eileenmcnaughton I need a help!!

How i can place the div section for button just besides the Save and Cancel buttom? Can i use JS to move the link or will i need to CRM/Core/Form/EntityForm.tpl to include sections insertions?

**Comment by pradpnayak** [comment]
@eileenmcnaughton I have updated the PR to include only form changes and moved installation code to an [extension](https://github.com/pradpnayak/org.civicrm.financialtypecustomfield).

File: templates/CRM/Financial/Form/FinancialType.tpl (hunk 1 above)
Comment (eileenmcnaughton) [reply templates/CRM/Financial/Form/FinancialType.tpl]: This seems ok?

screenshot 2018-07-24 03 39 32

Comment (pradpnayak) [reply templates/CRM/Financial/Form/FinancialType.tpl]: When you open in a new tab the 'View or Edit Financial Accounts' button goes below of Save and Cancel

**Comment by eileenmcnaughton** [comment]
@pradpnayak I tested this in view and edit mode and read through the code. I'm OK to merge this aside from your button alignment issue & I'm OK with that as a follow up too.

File: templates/CRM/Financial/Form/FinancialType.tpl (hunk 1 above)
Comment (eileenmcnaughton) [reply templates/CRM/Financial/Form/FinancialType.tpl]: Does overriding ```protected function addFormButtons()``` { help?

Comment (eileenmcnaughton) [reply templates/CRM/Financial/Form/FinancialType.tpl]: buttons look ok for me screenshot 2018-07-24 03 39 32

**Comment by eileenmcnaughton** [comment]
test this please

File: templates/CRM/Financial/Form/FinancialType.tpl (hunk 1 above)
Comment (eileenmcnaughton) [reply templates/CRM/Financial/Form/FinancialType.tpl]: hmm not sure maybe @colemanw has a good idea

I guess we could have an 'inner tpl' of

```
  {if $action eq 8}
    <div class="messages status no-popup">
      <div class="icon inform-icon"></div>
      {$deleteMessage|escape}
    </div>
  {else}
    <table class="form-layout-compressed">
      {foreach from=$entityFields item=fieldSpec}
        {assign var=fieldName value=$fieldSpec.name}
        <tr class="crm-{$entityInClassFormat}-form-block-{$fieldName}">
          {include file="CRM/Core/Form/Field.tpl"}
        </tr>
      {/foreach}
    </table>
    {include file="CRM/common/customDataBlock.tpl"}
  {/if}
```
that is included from CRM/Core/Form/EntityForm.tpl - so you can just include the inner if you want ... different buttons

**Comment by pradpnayak** [comment]
Thanks @eileenmcnaughton

**Pull request merged**